components/dashboard/sections/funds-and-grants/HostedGrants.tsx

```diff
@@ -0,0 +1,259 @@
+  const router = useRouter();
+  const intl = useIntl();
+  const pageRoute = `/dashboard/${hostSlug}/hosted-grants`;
+
+  const { data: metaData } = useQuery(
+    gql`
+      query HostedGrantsMetadata($hostSlug: String!) {
+        host(slug: $hostSlug) {
+          id
+          ...HostInfoCardFields
+          transferwise {
+            id
+            availableCurrencies
+            amountBatched {
+              valueInCents
+              currency
+            }
+          }
+        }
+        pending: expenses(host: { slug: $hostSlug }, status: [PENDING]) {
+          totalCount
+        }
+        approved: expenses(host: { slug: $hostSlug }, status: [APPROVED]) {
+          totalCount
+        }
+        paid: expenses(host: { slug: $hostSlug }, status: [PAID]) {
+          totalCount
+        }
+
+        hostedAccounts: accounts(host: { slug: $hostSlug }, orderBy: { field: ACTIVITY, direction: DESC }) {
+          nodes {
+            id
+            ...AccountHoverCardFields
+          }
+        }
+
+        expenseTags: expenseTagStats(host: { slug: $hostSlug }) {
+          nodes {
+            id
+            tag
+          }
+        }
+      }
+
+      ${accountHoverCardFields}
+      ${hostInfoCardFields}
+    `,
+    {
+      variables: { hostSlug, withHoverCard: true },
+      context: API_V2_CONTEXT,
+    },
+  );
+
+  const views: Views<FilterValues> = [
+    {
+      label: intl.formatMessage({ defaultMessage: 'All', id: 'zQvVDJ' }),
+      filter: {},
+      id: 'all',
+    },
+    {
+      label: intl.formatMessage({ defaultMessage: 'Awaiting approval', id: 'uPFUUe' }),
+      filter: { status: [ExpenseStatusFilter.PENDING], sort: { field: 'CREATED_AT', direction: 'ASC' } },
+      id: 'pending',
+      count: metaData?.pending?.totalCount,
+    },
+    {
+      label: intl.formatMessage({ defaultMessage: 'Approved', id: '6XFO/C' }),
+      filter: {
+        status: [ExpenseStatusFilter.APPROVED],
+        sort: { field: 'CREATED_AT', direction: 'ASC' },
+      },
+      id: 'approved',
+      count: metaData?.approved?.totalCount,
+    },
+    {
+      label: intl.formatMessage({ defaultMessage: 'Disbursed', id: 'eErSK0' }),
+      filter: { status: [ExpenseStatusFilter.PAID] },
+      count: metaData?.paid?.totalCount,
+      id: 'paid',
+    },
+  ];
+
+  const meta: FilterMeta = {
+    currency: metaData?.host?.currency,
+    hostSlug: hostSlug,
+    hostedAccounts: metaData?.hostedAccounts.nodes,
+    expenseTags: metaData?.expenseTags.nodes?.map(t => t.tag),
+    includeUncategorized: true,
+  };
+
+  const queryFilter = useQueryFilter({
+    schema: filterSchema,
+    toVariables,
+    defaultFilterValues: views[1].filter,
+    filters,
+    meta,
+    views,
+  });
+
+  const variables = {
+    hostSlug,
+    ...queryFilter.variables,
+  };
+
+  const expenses = useQuery(hostDashboardExpensesQuery, {
+    variables,
+    context: API_V2_CONTEXT,
+  });
+
+  const { data, error, loading } = expenses;
```

Comment:
We should refetch this query when a grant changes state.

Reply:
The grant state change already reflects on the list due to graphql cache update